spharpy/spherical.py

```diff
@@ -482,11 +530,18 @@ def spherical_harmonic_basis_gradient_real(n_max, coords):
     grad_phi : ndarray, float
         Gradient with respect to the azimuth angle.
 
-    """
-    n_points = coords.csize
-    n_coeff = (n_max+1)**2
-    theta = coords.colatitude
-    phi = coords.azimuth
+    """  # noqa: 501
+    if not isinstance(coordinates, pf.Coordinates):
+        axis = np.where(coordinates.shape == 3)[0][0]
+        if axis == 0:
+            coordinates = coordinates.T
+        coordinates = pf.Coordinates(coordinates[:, 0],
+                                     coordinates[:, 1],
+                                     coordinates[:, 2])
```

Comment:
Same comment as above. I don't think we should do auto-transforming coordinates.,

Reply:
Same as above. It is removed now.